src/main/java/uk/gov/digital/ho/hocs/document/MSGDocumentConverter.java

```diff
@@ -0,0 +1,216 @@
+public class MSGDocumentConverter {
+
+    private static final int PDF_WIDTH = 595;
+    private static final int PDF_HEIGHT = 842;
+
+    private static final String MSG_EXT = "msg";
+
+    private static final String TIF_EXT = "tif";
+    private static final String JPG_EXT = "jpg";
+    private static final String GIF_EXT = "gif";
+    private static final String PNG_EXT = "png";
+
+    private static final String TIFF_EXT = "tiff";
+    private static final String JPEG_EXT = "jpeg";
+
+    private static final String[] SUPPORTED_EXTENSIONS = { MSG_EXT };
+
+
+    boolean isSupported(String fileExtension) {
+        if (StringUtils.isEmpty(fileExtension)) {
+            return false;
+        }
+        final String extension = fileExtension.trim();
+        for (String ext : SUPPORTED_EXTENSIONS) {
+            if (ext.equalsIgnoreCase(extension)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    byte[] convertToPdf(String ext, InputStream inputStream) throws IOException, DocumentException {
+        if (MSG_EXT.equalsIgnoreCase(ext)) {
+            return convertMsg(inputStream);
+        }
+        throw new DocumentException("Unsupported format for conversion");
+    }
+
+
+    private byte[] convertMsg(InputStream inputStream) throws IOException, DocumentException {
+
+        MsgParser parser = new MsgParser();
+        Message message = parser.parseMsg(inputStream);
+        MsgContents contents = extractContents(message);
+
+        Document pdf = new Document();
+        final ByteArrayOutputStream arrayOutputStream = new ByteArrayOutputStream();
+        PdfWriter.getInstance(pdf, arrayOutputStream);
+        pdf.open();
+        pdf.newPage();
+
+        pdf.add(new Paragraph(String.format("From: %s [%s]", contents.getFromEmail(),
+                                                 StringUtils.isEmpty(contents.getFromName()) ? "N/A" : contents.getFromName())));
+        pdf.add(new Paragraph(MessageFormat.format("To: {0}", StringUtils.isEmpty(contents.getToEmail()) ?
+                                                                   contents.getToName() : contents.getToEmail())));
+        pdf.add(new Paragraph("Subject: " + contents.getSubject()));
+
+        if (!parseElements(pdf, contents)) {
+            pdf.add(new Paragraph(""));
+            pdf.add(new Paragraph(contents.getBodyText()));
+        }
+        for (Attachment attachment : contents.getAttachments()) {
+            processAttachment(pdf, attachment);
+        }
+        pdf.close();
+
+        final byte[] outputBytes = arrayOutputStream.toByteArray();
+        arrayOutputStream.close();
+        return outputBytes;
+    }
+
+    private MsgContents extractContents(Message message) {
+        MsgContents contents = new MsgContents();
+        contents.setFromEmail(message.getFromEmail());
+        contents.setFromEmail(contents.getFromEmail().contains("Content_Types") ? StringUtils.EMPTY : contents.getFromName());
```

Comment:
two lines (111 and 112 both setting the same property).

Reply:
Read it more carefully Hubert.